SPEC/BITSWAP.md

```diff
@@ -5,60 +5,59 @@
 
 ### `bitswap.wantlist`
 
-> Returns the wantlist, optionally filtered by peer ID
+> Get the list of wanted CIDs for this peer or another peer on the network.
 
-#### `Go` **WIP**
+#### Go **WIP**
 
-#### `JavaScript` - ipfs.bitswap.wantlist([peerId], [callback])
+#### JavaScript - `ipfs.bitswap.wantlist([peerId])`
 
-`callback` must follow `function (err, list) {}` signature, where `err` is an error if the operation was not successful. `list` is an Object containing the following keys:
+##### Parameters
 
-- `Keys` An array of objects containing the following keys:
-    - `/` A string multihash
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | `String` | (optional) Base 58 encoded Peer ID to get the wantlist for. Default: this node's peer ID |
 
-If no `callback` is passed, a promise is returned.
+##### Returns
 
-**Example:**
+| Type | Description |
+|------|-------------|
+| `Promise<{`<br/>&nbsp;&nbsp;`Keys<Array<Object>>`<br/>`}>` | The list of CIDs wanted by the peer. Each object in the array has a single property "/" a string CID. |
```

Comment:
this seems weird, html inside inline code block just to add new lines and space

Reply:
I know 😢. I would like this to be clearer when viewed as markdown but I don't know how right now. Unfortunately multi-line code blocks in a table aren't possible. If you have any ideas for how to format this in markdown so that it reads clearly when rendered then please comment!

Reply:
I've recently also started to use tables for parameters (for js-ipld). But I'll move away from that again. I find tables hard to read if you just read the unrendered Markdown.

The signature is quite complicated, I actually missed that it is an object containing a single field called `Keys`. I found that clearer in the old docs.

Reply:
> I've recently also started to use tables for parameters (for js-ipld). But I'll move away from that again. I find tables hard to read if you just read the unrendered Markdown.

👍

Reply:
Ok, I will think up a better way to layout this information that is readable rendered or not.